### PR TITLE
bug: `PhpdocToCommentFixer` support for enum cases

### LINKED ISSUE #6204: Wrong comment style in php 8.1 enums with phpdoc_to_comment rule
## Bug report

   * PHP version: 8.1.1
   * PHP CS Fixer version: 3.4.0
   * the command used to run PHP CS Fixer (run with `-vvv`):  
     * `vendor/bin/php-cs-fixer fix --dry-run --diff --show-progress=dots --allow-risky=yes`
   * the configuration (file)
```php
$finder = PhpCsFixer\Finder::create();
return (new PhpCsFixer\Config())
    ->setRules([
        '@PSR12' => true,
        '@PSR12:risky' => true,
        '@DoctrineAnnotation' => true,
        '@PHP74Migration' => true,
        '@PHP74Migration:risky' => true,
        '@PhpCsFixer' => true,
        '@PhpCsFixer:risky' => true,

        'fopen_flags' => [
            'b_mode' => true,
        ],

        'array_syntax' => [
            'syntax' => 'short',
        ],
        'yoda_style' => [
            'equal' => false,
            'identical' => false,
            'less_and_greater' => false,
        ],
        'increment_style' => [
            'style' => 'post',
        ],

        'cast_spaces' => [
            'space' => 'none',
        ],
        'concat_space' => [
            'spacing' => 'one',
        ],

        'multiline_whitespace_before_semicolons' => [
            'strategy' => 'no_multi_line',
        ],
        'method_chaining_indentation' => false,
        'method_argument_space' => false,
        'no_space_around_double_colon' => false,

        'class_definition' => [
            'multi_line_extends_each_single_line' => true,
        ],

        'single_line_throw' => false,
        'static_lambda' => true,
        'return_assignment' => false,
        'no_unset_on_property' => false,

        'no_superfluous_phpdoc_tags' => [
            'allow_mixed' => true,
            'remove_inheritdoc' => true,
        ],
        'phpdoc_to_comment' => [
            'ignored_tags' => [
                'psalm-suppress',
            ]
        ],
        'phpdoc_no_alias_tag' => [
            'replacements' => [],
        ],
        'phpdoc_add_missing_param_annotation' => false,
        'phpdoc_separation' => false,
        'phpdoc_align' => [
            'align' => 'left',
        ],

        'php_unit_strict' => false,
        'php_unit_test_case_static_method_calls' => [
            'call_type' => 'this',
        ],
    ])
    ->setFinder($finder);
```
-->

### Code snippet that reproduces the problem
Input:
```php
<?php
/**
 * Comment
 */
enum SomeEnum: string
{
    /**
     * Comment
     */
    case Name1 = 'name1';

    /**
     * Comment
     */
    case Name2 = 'name2';

    public function foo(): void
    {
    }
}
```

Output (`phpdoc_to_comment`, `no_extra_blank_lines`, `single_line_comment_style`):
```php
<?php
// Comment
enum SomeEnum: string
{
    // Comment
    case Name1 = 'name1';
    // Comment
    case Name2 = 'name2';
    public function foo(): void
    {
    }
}
```

### LINKED ISSUE #7030: PhpdocToCommentFixer false positives with enums
### Description

I have an enum type and want to deprecate some of the cases.

PhpdocToCommentFixer however wants to change these docblocks to regular comments.

### Runtime version

v3.16.0

### Used command

I'm using php-cs-fixer through symplify/easy-coding-standard so this is irrelevant.

### Configuration file

I'm using php-cs-fixer through symplify/easy-coding-standard so this is irrelevant.

### Code snippet that reproduces the problem

```php
enum MyEnum: string
{
    case A = 'a';

    case B = 'b';

    /**
     * @deprecated use D instead
     */
    case C = 'c';

    case D = 'd';
}
```

### PR DESCRIPTION
**Pull request by Wirone**
Fixes #7030 
Fixes #6204

### CODE REVIEW

**Review by kubawerlos — CHANGES_REQUESTED** [review]
Let's have [this](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7041) merged before to prevent regression.

**Comment by paulbalandan** [comment]
I think this should also fix #6204

File: src/Tokenizer/Analyzer/CommentsAnalyzer.php [hunk 1]
```diff
@@ -194,6 +194,12 @@ private function isStructuralElement(Tokens $tokens, int $index): bool
             return true;
         }
 
+        if ($token->isGivenKind(T_CASE) && \defined('T_ENUM')) {
+            $caseParent = $tokens->getPrevTokenOfKind($index, [[T_ENUM], [T_SWITCH]]);
+
+            return null === $caseParent || $tokens[$caseParent]->isGivenKind([T_ENUM]);
```
Comment (kubawerlos) [inline src/Tokenizer/Analyzer/CommentsAnalyzer.php:200]: Can `$caseParent` ever be `null`? If, so let's have a test to cover such a case.

Comment (Wirone) [reply src/Tokenizer/Analyzer/CommentsAnalyzer.php:200]: I've added that based on `getPrevTokenOfKind()`'s return type, but in practice there's no chance for that 😅. I'll remove this condition.

**Comment by Wirone** [comment]
Github Actions work weird right now, I force-pushed rebased branch 2 times and both workflows cancelled 1 of the jobs (and can't be retried). I'll try again tomorrow.

**Review by kubawerlos — APPROVED** [review]

**Pull request merged**